js/source/geojson_source.js

```diff
@@ -1,28 +1,23 @@
 'use strict';
 
-var util = require('../util/util');
 var Evented = require('../util/evented');
-var TilePyramid = require('./tile_pyramid');
-var Source = require('./source');
+var util = require('../util/util');
 var urlResolve = require('resolve-url');
 var EXTENT = require('../data/bucket').EXTENT;
 
-module.exports = GeoJSONSource;
+module.exports.create = function (id, options, dispatcher) {
+    return new GeoJSONSource(id, options, dispatcher);
+};
```

Comment:
Ah, shoot -- this was here because I had seen https://github.com/mapbox/mapbox-gl-js/issues/2506 with (a) no real objections and (b) marked on the v1.0 milestone... but @lucaswoj you mentioned in Slack that you'd rather wait and make that change wholesale, and I just lost track of the task of reverting to the `new`-based API here.